src/main/java/com/google/sps/data/Section.java

```diff
@@ -15,56 +15,101 @@
-    private List<TimeRange> meetingTimes;
-
-    public Section(String professor, List<TimeRange> meetingTimes){
-        this.professor = professor;
-        this.meetingTimes = meetingTimes;
-        Collections.sort(this.meetingTimes, TimeRange.ORDER_BY_START);
-
-        for(int i = 0; i < meetingTimes.size() - 1; i++){
-          if(meetingTimes.get(i).overlaps(meetingTimes.get(i + 1))) {
-            throw new IllegalArgumentException("A section's lecture times can't overlap!");
-          }
-        }
-    }
+  private String professor;
+  // Meeting times will be sorted in ascending order of time,
+  // with the first lesson being first in the list
+  private List<TimeRange> meetingTimes;
 
-    @Override
-    public boolean equals(Object other) {
-      if (other == this) {
-        return true;
-      }
+  public Section(String professor, List<TimeRange> meetingTimes) {
+    this.professor = professor;
+    this.meetingTimes = meetingTimes;
+    Collections.sort(this.meetingTimes, TimeRange.ORDER_BY_START);
 
-      if (!(other instanceof Section)) {
-        return false;
+    for (int i = 0; i < meetingTimes.size() - 1; i++) {
+      if (meetingTimes.get(i).overlaps(meetingTimes.get(i + 1))) {
+        throw new IllegalArgumentException("A section's lecture times can't overlap!");
       }
+    }
+  }
 
-      Section otherSection = (Section) other;
-
-      return otherSection.professor.equals(this.professor)
-              && otherSection.meetingTimes.equals(this.meetingTimes);
+  @Override
+  public boolean equals(Object other) {
+    if (other == this) {
+      return true;
     }
 
-    @Override
-    public int hashCode() {
-        return Objects.hash(professor, meetingTimes);
+    if (!(other instanceof Section)) {
+      return false;
     }
 
+    Section otherSection = (Section) other;
+
+    return otherSection.professor.equals(this.professor) 
+      && otherSection.meetingTimes.equals(this.meetingTimes);
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hash(professor, meetingTimes);
+  }
+
+  public List<TimeRange> getMeetingTimes() {
+    return this.meetingTimes;
+  }
 
-    /**
-     * Function to return whether or not a section overlaps with another
-     * @param other
-     * @return If there's overlap between the current section and the other section
-     */
-    public boolean overlaps(Section other){
-        //TODO: This
-        return false;
+  /**
+   * Function to return whether or not a section overlaps with another.
+   * 
+   * @param other the section to check for overlap with.
+   * @return If there's overlap between the current section and the other section
+   */
+  public boolean overlaps(Section other) {
+    LinkedList<TimeRange> thisTimes = new LinkedList<>(this.meetingTimes);
+    LinkedList<TimeRange> otherTimes = new LinkedList<>(other.getMeetingTimes());
+
+    TimeRange thisCurr = thisTimes.removeFirst();
+    TimeRange otherCurr = otherTimes.removeFirst();
```

Comment:
You should check for the case where one or both `TimeRange` lists are empty before fetching their respective first elements

Reply:
Added!